src/analyses/accessAnalysis.ml

```diff
@@ -1,156 +1,43 @@
-(** Access and data race analysis. *)
+(** Access analysis. *)
 
-module M = Messages
 module LF = LibraryFunctions
 open Prelude.Ana
 open Analyses
 open GobConfig
 
 
-(** Access and data race analyzer without base --- this is the new standard *)
+(** Access analyzer without base --- this is the new standard *)
 module Spec =
 struct
   include Analyses.DefaultSpec
 
   let name () = "access"
 
-  module D = Lattice.Unit
+  module D = AccessDomain.EventSet
```

Comment:
Are we worried about this set becoming large? If not what prevents it from growing for large programs?
Naively, it would seem `leq` checks for this EventSet would be quadratic in it's size?

Reply:
It's actually not very accumulating because `sync` empties it between transfer functions, such that the local sets would only have the set of accesses of just the last transfer function. I'll add a few comments to point out this unusual behavior.

Also, we already put these accesses into global access/race invariants, so we already have the same amount around, just stored in a different place.

Ideally it would suffice to collect these only during postsolving, but local constraint variable values cannot be modified during postsolving (unlike write-only globals), so they'd only emit fixpoint not reached errors and not persist. That's the same reason why the GraphML witness data collection happens all the time as well, not just during postsolving.

Reply:
I did an SV-COMP run with this and sadly there's ~15% average slowdown from these changes. I think it might be a bit about the sets, but more because now access events are emitted also in singlethreaded mode.

Reply:
I implemented some optimizations now:
1. Per-node accesses are only collected if they're enabled for _YAML_ witness generation (which we currently don't use for SV-COMP). Otherwise the old behavior remains: access events are only emitted in multithreaded mode.
2. Instead of collecting accesses into local states all the time, they're now collected only during postsolving into new write-only global constraint variables. This means they are not available during the analysis itself, but that wasn't required to begin with.

With these, performance on sv-benchmarks is on par with what it was before this PR.